### PR TITLE
Add Enum.extraNamesToValuesMap method

### PR DESCRIPTION
**Pull request by agustafson**
This is a PR for issue #321 

In order to map legacy string values to an `Enum`, I added a `Enum.extraNamesToValuesMap` which allows the user to define extra mappings.

Example from test:
```scala
sealed trait Result extends EnumEntry
object Result extends Enum[Result] {
  case object Good extends Result
  case object Bad  extends Result

  override val extraNamesToValuesMap: Map[String, Result] = Map[String, Result]("Ok" -> Good)

  override val values = findValues
}
```

In the case above, `Result.withName("Ok")` will return `Result.Good`.

### CODE REVIEW

**Review by lloydmeta — APPROVED** [review]
LGTM, just pending some things I want to fix on CI before merging.

**Comment by lloydmeta** [comment]
Thanks for this. Just released `1.7.0` with this.

**Comment by agustafson** [comment]
Great! Thanks so much for the speedy merge and release @lloydmeta - much appreciated!

And it's a great library too btw.

**Pull request merged**
